### PR TITLE
Configure Docker container for Fog Overseer

### PR DESCRIPTION
**Pull request by samdealy**
### Motivation
As described in #1482, we want to deploy Fog Overseer to CD. The first step is to containerize the service using Docker. 

### In this PR
* Creates a "minimal runtime container" for Fog Overseer. This means that the overseer binary is built outside the container and copied over into the container. 
* Modifies overseer configs to accept environment variables. This enables us to set overseer configs via helm charts.

#1482 

### Future Work
* Publish Docker image to Docker Hub.
* Create K8 manifests and helm charts for Fog Overseer container.

### CODE REVIEW

File: fog/overseer/server/src/config.rs [hunk 1]
```diff
@@ -11,20 +11,15 @@ use structopt::StructOpt;
 #[derive(Clone, Serialize, StructOpt)]
 pub struct OverseerConfig {
     /// Host to listen on.
-    #[structopt(long, default_value = "127.0.0.1")]
+    #[structopt(long, env, default_value = "127.0.0.1")]
```
Comment (wjuan-mob) [inline fog/overseer/server/src/config.rs:14]: If we intend to use this as an environment variable, does it make sense to give it a more specific name to make it less likely to conflict?

**Review by wjuan-mob — APPROVED** [review]

File: fog/overseer/server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM ubuntu:20.04
```
Comment (wjuan-mob) [inline fog/overseer/server/Dockerfile:1]: I think we normally specify a specific sha for this. Looking at the other dockerfiles we have in fullservice or mobilecoin.

File: fog/overseer/server/src/config.rs (hunk 1 above)
Comment (samdealy) [reply fog/overseer/server/src/config.rs:14]: sgtm, done.

**Review by jgreat — DISMISSED** [review]
Some changes for consistency.  We should where possible match artifacts with code.  cargo package = mc-fog-overseer-server, then all the downstream objects generated (docker container, docker repo, helm chart) should match so its easy to track back objects from deployment all the way back up to their code.

File: fog/overseer/server/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+RUN apt-get update \
+  && apt-get upgrade -y \
+  && apt-get clean \
+  && rm -r /var/lib/apt/lists
+
+WORKDIR /
+COPY bin/start_overseer_server /usr/local/bin/start_overseer_server
+
+# The default port that the Fog Overseer HTTP Server runs on.
+ARG DEFAULT_PORT=9090
```
[the remark this answers is not in the document]
Comment (samdealy) [reply fog/overseer/server/Dockerfile:11]: Thanks for the info. Removed the ARG as well as the OVERSEER_LISTEN_PORT ENV.

File: fog/overseer/server/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
+FROM ubuntu:20.04
+RUN apt-get update \
+  && apt-get upgrade -y \
+  && apt-get clean \
+  && rm -r /var/lib/apt/lists
+
+WORKDIR /
+COPY bin/start_overseer_server /usr/local/bin/start_overseer_server
```
[the remark this answers is not in the document]
Comment (samdealy) [reply fog/overseer/server/Dockerfile:8]: I just renamed it to "fog_overseer_server" to fit this convention. (IIUC we don't prefix the binary name with "mc". At least, we don't for Fog Ingest).

[the remark this answers is not in the document]
Comment (jcape) [reply fog/overseer/server/Dockerfile:8]: This needs to be renamed as well, otherwise this LGTM.

[the remark this answers is not in the document]
Comment (samdealy) [reply fog/overseer/server/Dockerfile:8]: It was renamed in this commit: https://github.com/mobilecoinfoundation/mobilecoin/pull/1503/commits/1343948f26550aa2872d3ebda0d6059006194616

[the remark this answers is not in the document]
Comment (samdealy) [reply fog/overseer/server/Dockerfile:8]: NVM, I was looking at the wrong line.

**Review by remoun — APPROVED** [review]

**Pull request merged**